stack-deploy/src/types.rs

```diff
@@ -544,3 +547,90 @@ impl Template {
         }
     }
 }
+
+#[derive(Debug)]
+pub struct TemplateBody(String);
+
+impl AsRef<[u8]> for TemplateBody {
+    fn as_ref(&self) -> &[u8] {
+        self.0.as_bytes()
+    }
+}
+
+impl From<String> for TemplateBody {
+    fn from(value: String) -> Self {
+        Self(value)
+    }
+}
+
+impl From<TemplateBody> for String {
+    fn from(value: TemplateBody) -> Self {
+        value.0
+    }
+}
+
+impl From<&TemplateBody> for String {
+    fn from(value: &TemplateBody) -> Self {
+        value.0.clone()
+    }
+}
+
+impl From<&str> for TemplateBody {
+    fn from(value: &str) -> Self {
+        value.to_string().into()
+    }
+}
+
+impl From<&TemplateBody> for aws_sdk_s3::primitives::ByteStream {
+    fn from(value: &TemplateBody) -> Self {
+        value.0.as_bytes().to_vec().into()
+    }
+}
+
+impl TemplateBody {
+    pub fn needs_upload(&self) -> bool {
+        self.0.len() > 51200
+    }
+}
+
+#[derive(Debug)]
+pub enum TemplateFormat {
+    JSON,
+    YAML,
+}
+
+impl TemplateFormat {
+    pub fn file_ext(&self) -> &str {
```

Comment:
__suggestion: (nitpick)__ This type could be changed to:

```patch
diff --git a/stack-deploy/src/types.rs b/stack-deploy/src/types.rs
index 287482e..eab9874 100644
--- a/stack-deploy/src/types.rs
+++ b/stack-deploy/src/types.rs
@@ -608,7 +608,7 @@ pub enum TemplateFormat {
 }
 
 impl TemplateFormat {
-    pub fn file_ext(&self) -> &str {
+    pub fn file_ext(&self) -> &'static str {
         match self {
             Self::JSON => "json",
             Self::YAML => "yaml",
```

Reply:
I think the lifetime elision does exactly the same, so I usually do not want extra syntax when there is no reason to do so semantically.